.github/workflows/cd.yml

```diff
@@ -0,0 +1,78 @@
+name: Continuous Deployment
+
+on:
+  push:
+    branches: [main]
+  workflow_dispatch:
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group:  ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pre-commit:
+    name: ruff and hooks.
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: '3.8'
+      - uses: pre-commit/action@v3.0.1
+        with:
+          extra_args: --all-files --show-diff-on-failure
+
+  cd:
+    name: CD
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        lfs: false
+
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.8'
```

Comment:
How about testing in CD for more than only the oldest Python version we currently support? Ideally all but we could also limit it.

Reply:
I would argue against this since the purpose of CD is not a (max. board) compatibility test. One could do this in the CI workflow as it is preferable to let these things fail fast there. Doing it at both places just doubles the effort without a real gain. It is highly likely that if sth is failing due to the wrong Python version, it is affecting the "not slow" tests.

Reply:
I don't think we should do that in the CD workflow as its purpose is a bit of a different one (computing max coverage, doc, etc.). Moreover, I think that it is very unlikely that there is a failure introduced due to Python version incompatibility which only occurs in the slow tests and not the others.

Reply:
I think @michaeldeistler had a strong opinion against increasing the runtime of our CI. In theory (with unlimited runners) adding more Python versions to CI will not increase the runtime as they're running in parallel but in cases like the hackathon this can make a difference. I don't have a strong opinion where to do it and on how many versions but I would be in favor of testing more than just one version somewhere.

Github Actions also has the option to schedule Actions like cron jobs, e.g. to run on multiple versions at least once per day.

Reply:
I've added a workflow that can be run manually to check all tests (or another marker) if desired